### PR TITLE
[REF-2168] Add share options to custom component commands

### PR DESCRIPTION
**Pull request by martinxu9**
## Summary
- Use tomlkit to process pyproject.toml, when publishing, ask user if want to interactively update the info (such as source URL, author email)
- Add a `share` flag for the `publish` command. By default after a successful publish, the command ask the user if they want to share information such as demo app link. This information goes to Reflex backend services.
- Add a `share` command so the same share logic can happen independent of the publish.
## Tests
- [x] Tested locally together with https://github.com/reflex-dev/reflex-hosting/pull/240 
## How to test
- Use the reflex in this PR. Use an existing custom component project or follow the docs to create a new one. The actual implementation does not matter. 
- Point to control plane dev by setting
```
export CP_WEB_URL=https://control-plane.dev.reflexcorp.run
export CP_BACKEND_URL=https://rxcp-dev-control-plane.fly.dev
```
- Run `reflex component share` (it also works when not at the top level of the project, user will be prompted to enter name)
- Answer all the questions (feel free to leave out the optional ones)
- Finally expect to see a new entry on the DB (if you have supabase dev access, look at the `customcomponent` table)
- Expect to see a new entry when querying the public endpoint. One way to easily do that is through swagger https://rxcp-dev-control-plane.fly.dev/docs#/default/get_custom_components_gallery_data_custom_components_gallery_get

### CODE REVIEW

File: reflex/custom_components/custom_components.py
Comment (masenf) [inline reflex/custom_components/custom_components.py:691]:
```suggestion
            "If you decide to do this later, you can run `reflex component share` command. Thank you!"
```

File: pyproject.toml
Comment (picklelo) [inline pyproject.toml:65]: Is this a new dependency or already included by one of the packages? Is there a reason for tomlkit rather than the base toml library?

**Review by picklelo — COMMENTED** [review]
Tested the workflow with my image-zoom app and it seemed to work well. I see it entered in the dev database.

Will wait for masen for final approval

File: pyproject.toml
Comment (martinxu9) [reply pyproject.toml:65]: tomllib was introduced in 3.11 and it intentionally only includes read/parse. There's no write.

Comment (martinxu9) [reply pyproject.toml:65]: tomllib was introduced in 3.11 and it intentionally only includes read/parse. There's no write.

**Review by masenf — APPROVED** [review]
i like the toml parsing much better. nice

i wasn't able to actually test it with my new library yet, due to deploy issues. but the code looks good.

**Comment by martinxu9** [comment]
> i like the toml parsing much better. nice
> 
> 
> 
> i wasn't able to actually test it with my new library yet, due to deploy issues. but the code looks good.

I'll try to invite the whole team to share and seed our db for the gallery

**Review by masenf — APPROVED** [review]

**Review by Lendemor — APPROVED** [review]

**Pull request merged**
